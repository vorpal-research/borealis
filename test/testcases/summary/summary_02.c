#include <stdio.h>
#include <stdlib.h>

// @requires \is_valid_ptr(arr)
void another_malloc(int** arr, size_t size, int safe) {
    int* res = malloc(sizeof(int) * size);
    if (safe > 0) {
        if (res)
            *arr = res;
        else
            exit(255);
    } else {
        *arr = res;
    }
}

void my_free(int* arr) {
    if (arr != NULL) {
        free(arr);
    }
}

int main() {
    int* arr;
    another_malloc(&arr, 16, 1);

    for (int i = 0; i < 16; ++i) {
        arr[i] = i;
    }
    int sixteen = arr[15];
    my_free(arr);

    return sixteen;
}
