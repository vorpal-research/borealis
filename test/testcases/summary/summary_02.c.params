---summary:sum-mode:interpol
---analysis:craig-colton-bounds:on
