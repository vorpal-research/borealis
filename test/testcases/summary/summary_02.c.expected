[
	{
		"location" : 
		{
			"filename" : "test/testcases/summary/summary_02.c",
			"loc" : 
			{
				"col" : 9,
				"line" : 28
			}
		},
		"type" : "BUF-01"
	},
	{
		"location" : 
		{
			"filename" : "test/testcases/summary/summary_02.c",
			"loc" : 
			{
				"col" : 5,
				"line" : 30
			}
		},
		"type" : "BUF-01"
	},
	{
		"location" : 
		{
			"filename" : "test/testcases/summary/summary_02.c",
			"loc" : 
			{
				"col" : 9,
				"line" : 28
			}
		},
		"type" : "INI-03"
	}
]
